.gitignore

```diff
@@ -1,9 +1,137 @@
-coverage
-node_modules
-test/registry/couch
-test/registry/data
-test/tmp
+
+# Created by https://www.gitignore.io/api/macos,windows,linux,node
+
+### Linux ###
+*~
+
+# temporary files which can be created if a process still has a handle open of a deleted file
+.fuse_hidden*
+
+# KDE directory preferences
+.directory
+
+# Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# .nfs files are created when an open file is removed but is still being accessed
+.nfs*
+
+### macOS ###
+*.DS_Store
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
+
+# Thumbnails
+._*
+
+# Files that might appear in the root of a volume
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+.com.apple.timemachine.donotpresent
+
+# Directories potentially created on remote AFP share
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+
+### Node ###
+# Logs
+logs
 *.log
-*.dump
-.DS_Store
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# Runtime data
+pids
+*.pid
+*.seed
+*.pid.lock
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
+
+# Coverage directory used by tools like istanbul
+coverage
+
+# nyc test coverage
 .nyc_output
+
+# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
+
+# Bower dependency directory (https://bower.io/)
+bower_components
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (http://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# Typescript v1 declaration files
+typings/
+
+# Optional npm cache directory
+.npm
+
+# Optional eslint cache
+.eslintcache
+
+# Optional REPL history
+.node_repl_history
+
+# Output of 'npm pack'
+*.tgz
+
+# Yarn Integrity file
+.yarn-integrity
+
+# dotenv environment variables file
+.env
+
+
+### Windows ###
+# Windows thumbnail cache files
+Thumbs.db
+ehthumbs.db
+ehthumbs_vista.db
+
+# Folder config file
+Desktop.ini
+
+# Recycle Bin used on file shares
+$RECYCLE.BIN/
+
+# Windows Installer files
+*.cab
+*.msi
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
+# End of https://www.gitignore.io/api/macos,windows,linux,node
+
+package-lock.json
+yarn.lock
+
```

Comment:
ignoring lockfiles like this is harmful, because lockfiles are still in the repo (they're just **invisible**)
See https://github.com/luftywiranda13/remove-lockfiles#why

Reply:
What do you mean 'they're just invisible' ?
If they are not committed to the repo, they are not in the repo.

Reply:
Also I added [npmrc](https://github.com/semantic-release/semantic-release/pull/451/files#diff-794bdde32142176a26aef68e6770521eR1) and [.yarnrc](https://github.com/semantic-release/semantic-release/pull/451/files#diff-a3e03be55c12fe638fe1e1c5aa684b9aR1) to not regenerate them.

Reply:
i updated my comment, sorry 😆

Reply:
> Also I added npmrc and .yarnrc to not regenerate them.

2 dotfiles to prevent lockfiles to be generated? I think it's overkill

Reply:
The point of removing the lockfiles is to do the development with latest versions of dependencies. Your solution (unless I'm missing something) just remove the lockfiles before commit even though they wouldn't be committed anyway due to gitignore.

That doesn't solve the problem as the lockfile will be present on the local machine until the contributor do a commit. So a contributor would do all the development and run the test locally with potentially outdated libraries. Therefore what worked on their local machines might not work on the CI.

Also adding 2 extra dependency + a commit hook seem more overkill to me than two one line dotfiles.

Reply:
> The point of removing the lockfiles is to do the development with latest versions of dependencies.

yes, I know. that's the main reason why I decide to never commit lockfiles in my projects



> Also adding 2 extra dependency + a commit hook seem more overkill to me than two one line dotfiles.


hmmm I think https://github.com/semantic-release/semantic-release/pull/451#discussion_r141984300 is a relevant answer